### PR TITLE
Add tests for name accessors in GSInstance

### PR DESCRIPTION
**Pull request by IvanUkhov**
As discussed in #881, this PR adds more tests.

### CODE REVIEW

File: Lib/glyphsLib/classes.py [hunk 1]
```diff
@@ -3357,14 +3357,12 @@ def windowsLinkedToStyle(self):
 
     @property
     def fontName(self):
-        value = (
+        return (
```
Comment (IvanUkhov) [inline Lib/glyphsLib/classes.py:3360]: Just unifying with the other accessors being tested.

File: Lib/glyphsLib/classes.py [hunk 2]
```diff
@@ -3374,10 +3372,9 @@ def fontName(self, value):
 
     @property
     def fullName(self):
-        value = self.customParameters["postscriptFullName"]
-        if value:
-            return value
-        return self.familyName + " " + self.name
+        return self.customParameters["postscriptFullName"] or (
```
Comment (IvanUkhov) [inline Lib/glyphsLib/classes.py:3375]: The same here.

File: tests/builder/instances_test.py [hunk 3]
```diff
@@ -148,19 +216,19 @@ def test_glyphs3_instance_filtering():
...
 def test_glyphs3_instance_properties(tmpdir):
-    expected_num_properties = [0, 0, 1, 1, 1, 1]
+    expected_num_properties = [0, 0, 1, 2, 1, 2]
 
     file = "InstanceFamilyName-G3.glyphs"
     font = glyphsLib.GSFont(os.path.join(DATA, file))
 
-    for instance, num in zip(font.instances, expected_num_properties):
-        assert len(instance.properties) == num
+    for expected, instance in zip(expected_num_properties, font.instances):
```
Comment (IvanUkhov) [inline tests/builder/instances_test.py:224]: To have the expected value first for consistency.

File: tests/builder/instances_test.py [hunk 4]
```diff
@@ -148,19 +216,19 @@ def test_glyphs3_instance_filtering():
 
 
 def test_glyphs3_instance_properties(tmpdir):
-    expected_num_properties = [0, 0, 1, 1, 1, 1]
+    expected_num_properties = [0, 0, 1, 2, 1, 2]
```
Comment (IvanUkhov) [inline tests/builder/instances_test.py:219]: Enriched the test file for the other test.

File: Lib/glyphsLib/classes.py [hunk 5]
```diff
@@ -3357,14 +3357,12 @@ def windowsLinkedToStyle(self):
 
     @property
     def fontName(self):
-        value = (
+        return (
             self.properties.get("postscriptFontName")
             or self.customParameters["postscriptFontName"]
```
Comment (schriftgestalt) [inline Lib/glyphsLib/classes.py:3362]: I would convert all custom parameters on loading the file. That makes all the accessor code much simpler.

**Comment by IvanUkhov** [comment]
@anthrotype, here are the tests.

File: Lib/glyphsLib/classes.py (hunk 5 above)
Comment (anthrotype) [reply Lib/glyphsLib/classes.py:3362]: > I would convert all custom parameters on loading the file.

I don't disagree it'd be desirable, but upconverting glyphs v2 files to a v3 internal representation upon reading (and down-converting to v2 if requested upon writing) would involve some major refactoring.

**Pull request merged**